emcc.py

```diff
@@ -1898,10 +1898,7 @@ def phase_linker_setup(options, state, newargs):
     if not settings.PURE_WASI and '-nostdlib' not in newargs and '-nodefaultlibs' not in newargs:
       default_setting('STACK_OVERFLOW_CHECK', max(settings.ASSERTIONS, settings.STACK_OVERFLOW_CHECK))
 
-  if settings.LLD_REPORT_UNDEFINED or settings.STANDALONE_WASM:
-    # Reporting undefined symbols at wasm-ld time requires us to know if we have a `main` function
-    # or not, as does standalone wasm mode.
-    # TODO(sbc): Remove this once this becomes the default
+  if settings.STANDALONE_WASM:
```

Comment:
Also this?

(I'd expect this PR to only flip `LLD_REPORT_UNDEFINED` and nothing else?)

Reply:
This PR doesn't just enable LLD_REPORT_UNDEFINED but also fixes all the remaining issues with it.   If you like I can make a separate PR called "fix remaining LLD_REPORT_UNDEFINED issues" before landing this change?

Reply:
Yes, if that's not hard to do it sounds better? Or, writing up what those fixes are in this PR.

Reply:
Done.  This PR is now just flipping the default and updating some test expectations.

Reply:
I still see a diff on this line? lgtm aside from that.

Reply:
Ah yes, I fixed LLD_REPORT_UNDEFINED so we no longer need to force main to be defined.  That would break existing builds.

With this change we can turn on LLD_REPORT_UNDEFINED without requiring main function by default.

One day we can try to switch the IGNORE_MISSING_MAIN default to false, but for now we keep at at true so that this change is as NFC as possible.

Reply:
(The comment below is no longer true which is why I removed it)

Reply:
The required for `IGNORE_MISSING_MAIN` to be disabled in `LLD_REPORT_UNDEFINED` mode was removed in #17175, BTW, not this change.